app/javascript/src/components/Dashboard/Notes/NoteItem/NoteItem.jsx

```diff
@@ -1,50 +1,76 @@
-import React from "react";
+import React, { useState } from "react";
 
 import { Clock, MenuVertical } from "neetoicons";
-import { Button, Avatar, Tooltip, Typography } from "neetoui";
+import { Button, Avatar, Tooltip, Typography, Dropdown, Toastr } from "neetoui";
 import PropTypes from "prop-types";
 
-const NoteItem = ({ title, description, updatedAt, userName }) => (
-  <div className="mb-4 w-full border border-gray-200 bg-white p-6 shadow-md dark:border-gray-700 dark:bg-gray-800">
-    <div className="flex justify-between">
-      <h2>{title}</h2>
-      <MenuVertical color="#1e1e20" size={24} />
-    </div>
-    <p className="text-base">{description}</p>
-    <hr className="solid mt-5 mb-5" />
-    <div className="flex justify-between">
-      <Button
-        className="rounded-md border-solid border-gray-200 text-xs font-light"
-        label="Getting Started"
-        size="medium"
-        style="text"
-        type="button"
-        onClick={() => {}}
-      />
-      <div className="flex items-center">
-        <Clock color="#1e1e20" size={22} />
-        <Tooltip content="Wednesday,10:30am" position="bottom">
-          <Typography className="mx-2" style="body3">
-            {updatedAt}
-          </Typography>
-        </Tooltip>
-        <Avatar
+import DeleteAlert from "../DeleteAlert";
+
+const NoteItem = ({
+  id,
+  title,
+  description,
+  updatedAt,
+  userName,
+  notes,
+  setNotes,
+}) => {
+  const [open, setOpen] = useState(false);
+
+  const handleDelete = () => {
+    setNotes(notes.filter(note => note.id !== id));
+    setOpen(false);
+    Toastr.success("Note deleted successfully");
+  };
+  return (
+    <div className="mb-4 w-full border border-gray-200 bg-white p-6 shadow-md dark:border-gray-700 dark:bg-gray-800">
+      <div className="flex justify-between">
+        <h2>{title}</h2>
+        <Dropdown buttonSize="large" buttonStyle="text" icon={MenuVertical}>
+          <li>Edit</li>
+          <li onClick={() => setOpen(true)}>Delete</li>
+        </Dropdown>
+      </div>
+      <p className="text-base">{description}</p>
+      <hr className="solid mt-5 mb-5" />
+      <div className="flex justify-between">
+        <Button
+          className="rounded-md border-solid border-gray-200 text-xs font-light"
+          label="Getting Started"
           size="medium"
-          user={{
-            name: userName,
-          }}
+          style="text"
+          type="button"
           onClick={() => {}}
         />
+        <div className="flex items-center">
+          <Clock color="#1e1e20" size={22} />
+          <Tooltip content="Wednesday, 10:30AM" position="bottom">
+            <Typography className="mx-2" style="body3">
+              {updatedAt}
```

Comment:
I believe I had pointed out a similar mistake to you before. Here the Tooltip content and the value displayed as text in the UI doesn't match.

Reply:
I am having a doubt here. Aren't we supposed to hardcode the `Tooltip` content?

Reply:
Since most of the values in this project are hardcoded I thought this one should also follow the same. Other than that, there are no other specific reasons.

Reply:
@soorajbhskrn First we need to establish what is the purpose of this tooltip. Then only we can come to a conclusion on whether we need to hardcode it or not.

So from your code this is what I understand:
- We display `updatedAt` as text in a short date format.
- When we hover our mouse pointer over it, we show the same `updatedAt` date in a longer date format.
- So hardcoding won't work.

What's the issue with hardcoding the way you've done it now?
- The date displayed as text and the date displayed in tooltip won't match. Which creates confusion to the consumer.

Lemme know if I am misunderstanding something here regarding the use of this tooltip. Maybe you had a different idea in mind. Lemme know about it. If you can convey me the strong and sensible(from consumer's perspective obviously) reasons for hardcoding the date in the tooltip over here, then we can go with that.

Reply:
I will narrow down what exactly I meant by the above message. Since most of the contents inside this project were hardcoded I thought this should also follow the same fashion. So I avoided the hassle of formatting the date by bringing new utility functions since the project was meant to concentrate more on `neeto-UI`.

While building consumer-based applications this should never be followed as it contains a date which is dynamic in nature.
I have corrected these mistakes in my latest PR.
Sorry for the inconvenience and going forward I will make sure to re-verify these scenarios before actually raising a PR.